src/controlllers/event.controller.ts

```diff
@@ -0,0 +1,80 @@
+import * as EventService from '../services/events.service'
+import { NextFunction, Response } from 'express';
+import { IExpressRequest, } from '../interfaces';
+import * as ResponseManager from '../helpers/response.manager';
+
+
+export async function handleCreateEvent(req: IExpressRequest, res: Response): Promise<void> {
+  try {
+    req.body.creator = <string>req.userId;
+    const { name, description, creator, location, type, isDraft, date, endDate, isPublic } = req.body
+    const response = await EventService.createEvent(
+      { name, description, creator, location, type, isDraft, date, endDate, isPublic })
+    return ResponseManager.success(res, response);
+  } catch (error: unknown) {
+    ResponseManager.handleError(res, error)
+  }
+}
+
+
+export async function handleGetEvents(req: IExpressRequest, res: Response): Promise<void> {
+  try {
+    const { page, size, search } = req.query
+    const response = await EventService.getAllEvents({ page, size, search })
+    return ResponseManager.success(res, response);
+  } catch (error: unknown) {
+    ResponseManager.handleError(res, error)
+  }
+}
+
+export async function handleGetOneEvent(req: IExpressRequest, res: Response): Promise<void> {
+  try {
+    const eventId = <string>req.params.eventId
+    const response = await EventService.getOneEvent(eventId);
+    return ResponseManager.success(res, response);
+  } catch (error: unknown) {
+    ResponseManager.handleError(res, error)
+  }
+}
+
+export async function handleUpdateEvent(req: IExpressRequest, res: Response, next: NextFunction): Promise<void> {
+  try {
+    const { name, description, creator, location, type, isDraft, date, endDate, isPublic } = req.body
+    const filter = { _id: <string>req.params.eventId, creator: <string>req.userId }
+    const response = await EventService.updateEvent(filter, {
+      name,
```

Comment:
What happens when one of these values are not supplied?

Reply:
I'm not really sure, mongoose just leave the document as they were.